software/kallistobustools/ref/main.nf

```diff
@@ -0,0 +1,68 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process KALLISTOBUSTOOLS_REF {
+    tag "$meta.id"
+    label 'process_medium'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), publish_id:meta.id) }
+
+    conda (params.enable_conda ? "bioconda::kb-python=0.25.1" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/kb-python:0.25.1--py_0"
+    } else {
+        container "quay.io/biocontainers/kb-python:0.25.1--py_0"
+    }
+
+    input:
+    tuple   val(meta), path(fasta)
```

Comment:
Do we need `meta` here? i.e. will this index be created for all samples in which case we don't need to define `meta`

Reply:
No you are right, since this is an index I guess meta is not needed! It's good to know for me for any future modules. Might also be useful to mention this somewhere in the docs for new module developers that they can omit meta when creating the module... (Maybe it's written somewhere and I missed it!)